src/anyvar/restapi/main.py

```diff
@@ -0,0 +1,284 @@
+    "/locations/{location_id}",
+    response_model=GetSequenceLocationResponse,
+    summary="Retrieve sequence location",
+    description="Retrieve registered sequence location by ID",
+    tags=[EndpointTag.LOCATIONS]
+)
+def get_location_by_id(
+    request: Request,
+    location_id: StrictStr = Query(..., description="Location VRS ID")
+):
+    """Retrieve stored location object by ID.
+
+    :param request: FastAPI request object
+    :param location_id: VRS location identifier
+    :return: complete location object if successful
+    :raise HTTPException: if requested location isn't found
+    """
+    av: AnyVar = request.app.state.anyvar
+    try:
+        location = av.get_object(location_id)
+    except KeyError:
+        return HTTPException(status_code=HTTPStatus.NOT_FOUND)
+    if location:
+        return {"location": location.as_dict()}
+    else:
+        return {"location": None}
+
+
+@app.put(
+    "/variation",
+    response_model=RegisterVariationResponse,
+    summary="Register a new variation object",
+    description="Provide a variation definition to be normalized and registered with AnyVar. A complete VRS object and digest is returned for later reference.",  # noqa: E501
+    tags=[EndpointTag.VARIATIONS]
+)
+def register_variation(
+    request: Request,
+    variation: RegisterVariationRequest = Body(
+        description="Variation description, including (at minimum) a definition property"  # noqa: E501
+    )
+):
+    """Register a variation based on a provided description or reference.
+
+    :param request: FastAPI request object
+    :param variation: provided variation description
+    :return: messages describing translation failure, or object and references if
+        successful
+    """
+    av: AnyVar = request.app.state.anyvar
+    definition = variation.definition
+    result = {
+        "object": None,
+        "messages": []
+    }
+    try:
+        translated_variation = av.translator.translate(var=definition)
+    except TranslationException:
+        result["messages"].append(f"Unable to translate {definition}")
+    except NotImplementedError:
+        result["messages"].append(
+            f"Variation class for {definition} is currently unsupported."
+        )
+    else:
+        if translated_variation:
+            v_id = av.put_object(translated_variation)
+            result["object"] = translated_variation.as_dict()
+            result["object_id"] = v_id
+        else:
+            result["messages"].append(f"Translation of {definition} failed.")
+    return result
+
+
+@app.put(
+    "/vrs_variation",
+    summary="Register a VRS variation",
+    description="Provide a valid VRS variation object to be registered with AnyVar. A digest is returned for later reference.",  # noqa: E501
+    response_model=RegisterVrsVariationResponse,
+    tags=[EndpointTag.VARIATIONS]
+)
+def register_vrs_object(
+    request: Request,
+    variation: VrsVariation = Body(description="Valid VRS object.")
```

Comment:
The default request body gets a 422. This is a vrsatile-pydantic issue. We'll need to add some examples to the classes

Reply:
👍  we should probably provide our own example object in anyvar anyway